{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pymongo import MongoClient\n",
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "MONGO_HOST = \"34.222.190.54\"\n",
    "MONGO_PORT = \"27017\"\n",
    "MONGO_DB = \"icd11\"\n",
    "MONGO_USER = \"icd11_user\"\n",
    "MONGO_PASS = \"icd11_pwd\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "ename": "ConfigurationError",
     "evalue": "command SON([('saslStart', 1), ('mechanism', 'SCRAM-SHA-1'), ('autoAuthorize', 1), ('payload', Binary(b'n,,n=icd11_user,r=NDMzNDA2MjE0MDM0NzI3MzQ=', 0))]) on namespace admin.$cmd failed: Authentication failed.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mOperationFailure\u001b[0m                          Traceback (most recent call last)",
      "\u001b[0;32m~/anaconda2/envs/GenomX/lib/python3.6/site-packages/pymongo/mongo_client.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, host, port, max_pool_size, document_class, tz_aware, _connect, **kwargs)\u001b[0m\n\u001b[1;32m    391\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 392\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_cache_credentials\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msource\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcredentials\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_connect\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    393\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mOperationFailure\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mexc\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/GenomX/lib/python3.6/site-packages/pymongo/mongo_client.py\u001b[0m in \u001b[0;36m_cache_credentials\u001b[0;34m(self, source, credentials, connect)\u001b[0m\n\u001b[1;32m    466\u001b[0m                 \u001b[0;31m# there is no need to call it here.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 467\u001b[0;31m                 \u001b[0mauth\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mauthenticate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcredentials\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msock_info\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__simple_command\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    468\u001b[0m                 \u001b[0msock_info\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mauthset\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0madd\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcredentials\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/GenomX/lib/python3.6/site-packages/pymongo/auth.py\u001b[0m in \u001b[0;36mauthenticate\u001b[0;34m(credentials, sock_info, cmd_func)\u001b[0m\n\u001b[1;32m    474\u001b[0m     \u001b[0mauth_func\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_AUTH_MAP\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmechanism\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 475\u001b[0;31m     \u001b[0mauth_func\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcredentials\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msock_info\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcmd_func\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    476\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/GenomX/lib/python3.6/site-packages/pymongo/auth.py\u001b[0m in \u001b[0;36m_authenticate_default\u001b[0;34m(credentials, sock_info, cmd_func)\u001b[0m\n\u001b[1;32m    449\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0msock_info\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmax_wire_version\u001b[0m \u001b[0;34m>=\u001b[0m \u001b[0;36m3\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 450\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0m_authenticate_scram_sha1\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcredentials\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msock_info\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcmd_func\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    451\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/GenomX/lib/python3.6/site-packages/pymongo/auth.py\u001b[0m in \u001b[0;36m_authenticate_scram_sha1\u001b[0;34m(credentials, sock_info, cmd_func)\u001b[0m\n\u001b[1;32m    237\u001b[0m     _scram_sha1_conversation(credentials, sock_info, cmd_func,\n\u001b[0;32m--> 238\u001b[0;31m                              sasl_start, sasl_continue)\n\u001b[0m\u001b[1;32m    239\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/GenomX/lib/python3.6/site-packages/pymongo/auth.py\u001b[0m in \u001b[0;36m_scram_sha1_conversation\u001b[0;34m(credentials, sock_info, cmd_func, sasl_start, sasl_continue)\u001b[0m\n\u001b[1;32m    184\u001b[0m     \u001b[0msasl_start\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'payload'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mBinary\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"n,,\"\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mfirst_bare\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 185\u001b[0;31m     \u001b[0mres\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcmd_func\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msock_info\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msource\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msasl_start\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    186\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/GenomX/lib/python3.6/site-packages/pymongo/mongo_client.py\u001b[0m in \u001b[0;36m__simple_command\u001b[0;34m(self, sock_info, dbname, spec)\u001b[0m\n\u001b[1;32m    702\u001b[0m             repr(spec).replace(\"%\", \"%%\"), ns)\n\u001b[0;32m--> 703\u001b[0;31m         \u001b[0mhelpers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_command_response\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresponse\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmsg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    704\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mend\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mstart\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/GenomX/lib/python3.6/site-packages/pymongo/helpers.py\u001b[0m in \u001b[0;36m_check_command_response\u001b[0;34m(response, reset, msg, allowable_errors)\u001b[0m\n\u001b[1;32m    181\u001b[0m             \u001b[0mmsg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmsg\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0;34m\"%s\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 182\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mOperationFailure\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmsg\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0merrmsg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    183\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mOperationFailure\u001b[0m: command SON([('saslStart', 1), ('mechanism', 'SCRAM-SHA-1'), ('autoAuthorize', 1), ('payload', Binary(b'n,,n=icd11_user,r=NDMzNDA2MjE0MDM0NzI3MzQ=', 0))]) on namespace admin.$cmd failed: Authentication failed.",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mConfigurationError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-6-0ba6ff5887f9>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mclient\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mMongoClient\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"mongodb://icd11_user:icd11_pwd@34.222.190.54:27017\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mdb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mclient\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mMONGO_DB\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mdb\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcommand\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"serverStatus\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;32mexcept\u001b[0m \u001b[0mException\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"You are connected!\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/GenomX/lib/python3.6/site-packages/pymongo/mongo_client.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, host, port, max_pool_size, document_class, tz_aware, _connect, **kwargs)\u001b[0m\n\u001b[1;32m    392\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_cache_credentials\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msource\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcredentials\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_connect\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    393\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mOperationFailure\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mexc\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 394\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mConfigurationError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    395\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    396\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_cached\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdbname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcoll\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mConfigurationError\u001b[0m: command SON([('saslStart', 1), ('mechanism', 'SCRAM-SHA-1'), ('autoAuthorize', 1), ('payload', Binary(b'n,,n=icd11_user,r=NDMzNDA2MjE0MDM0NzI3MzQ=', 0))]) on namespace admin.$cmd failed: Authentication failed."
     ]
    }
   ],
   "source": [
    "client = MongoClient(\"mongodb://icd11_user:icd11_pwd@34.222.190.54:27017\")\n",
    "db = client(MONGO_DB)\n",
    "try: db.command(\"serverStatus\")\n",
    "except Exception as e: print(e)\n",
    "else: print(\"You are connected!\")\n",
    "client.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "mms_col = db[\"mms\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "DATA = []\n",
    "for x in mms_col.find():\n",
    "    DATA.append(x) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{u'_id': ObjectId('5d950eec0ef932347358fa6e'),\n",
       "  u'childs': [],\n",
       "  u'code': u'QC68',\n",
       "  u'defn': u'NA',\n",
       "  u'has_children': 0,\n",
       "  u'has_parents': 1,\n",
       "  u'index_term': [{u'indexTerm_foundation_id': u'NA',\n",
       "    u'indexTerm_title': u'Family history of consanguinity'}],\n",
       "  u'is_leaf_node': False,\n",
       "  u'mms_id': u'1192168487',\n",
       "  u'parents': [u'1143554706'],\n",
       "  u'postcoordinationScale': [],\n",
       "  u'title': u'Family history of consanguinity'},\n",
       " {u'_id': ObjectId('5d950eec0ef932347358fa6f'),\n",
       "  u'childs': [],\n",
       "  u'code': u'3B64.13',\n",
       "  u'defn': u'A disease caused by determinants such as a blood transfusion that lead to an immune response to the foreign antigens. This disease is characterised by low levels of platelets in the body due to an immune reactive response to the foreign platelet antigens. This disease may present with increased bruising or haemorrhaging. Confirmation is by identification of decreased platelet count and presence of autoantibodies in a blood sample.',\n",
       "  u'has_children': 0,\n",
       "  u'has_parents': 1,\n",
       "  u'index_term': [{u'indexTerm_foundation_id': u'NA',\n",
       "    u'indexTerm_title': u'Alloimmune thrombocytopenia'}],\n",
       "  u'is_leaf_node': False,\n",
       "  u'mms_id': u'251307303',\n",
       "  u'parents': [u'526155201'],\n",
       "  u'postcoordinationScale': [],\n",
       "  u'title': u'Alloimmune thrombocytopenia'},\n",
       " {u'_id': ObjectId('5d950eec0ef932347358fa70'),\n",
       "  u'childs': [],\n",
       "  u'code': u'ND14.A0',\n",
       "  u'defn': u'NA',\n",
       "  u'has_children': 0,\n",
       "  u'has_parents': 1,\n",
       "  u'index_term': [{u'indexTerm_foundation_id': u'NA',\n",
       "    u'indexTerm_title': u'Strain or sprain of metatarsophalangeal joint of lesser toe'}],\n",
       "  u'is_leaf_node': False,\n",
       "  u'mms_id': u'2074348881',\n",
       "  u'parents': [u'43054047'],\n",
       "  u'postcoordinationScale': [{u'allowMultipleValues': u'NotAllowed',\n",
       "    u'axisName': u'laterality',\n",
       "    u'requiredPostcoordination': u'false',\n",
       "    u'scaleEntity': [u'627678743',\n",
       "     u'271422288',\n",
       "     u'876572005',\n",
       "     u'1038788978',\n",
       "     u'1694310660']},\n",
       "   {u'allowMultipleValues': u'AllowAlways',\n",
       "    u'axisName': u'associatedWith',\n",
       "    u'requiredPostcoordination': u'false',\n",
       "    u'scaleEntity': [u'980862652',\n",
       "     u'851395624',\n",
       "     u'73322695',\n",
       "     u'128104623',\n",
       "     u'1491356379',\n",
       "     u'491063206',\n",
       "     u'185842968',\n",
       "     u'2143219175',\n",
       "     u'558785723',\n",
       "     u'other',\n",
       "     u'unspecified']}],\n",
       "  u'title': u'Strain or sprain of metatarsophalangeal joint of lesser toe'},\n",
       " {u'_id': ObjectId('5d950eec0ef932347358fa71'),\n",
       "  u'childs': [],\n",
       "  u'code': u'XA8YJ9',\n",
       "  u'defn': u'NA',\n",
       "  u'has_children': 0,\n",
       "  u'has_parents': 1,\n",
       "  u'index_term': [{u'indexTerm_foundation_id': u'NA',\n",
       "    u'indexTerm_title': u'Sigmoid colon'},\n",
       "   {u'indexTerm_foundation_id': u'NA',\n",
       "    u'indexTerm_title': u'Sigmoid large intestine'},\n",
       "   {u'indexTerm_foundation_id': u'NA', u'indexTerm_title': u'Pelvic colon'},\n",
       "   {u'indexTerm_foundation_id': u'NA', u'indexTerm_title': u'Sigmoid, NOS'},\n",
       "   {u'indexTerm_foundation_id': u'NA',\n",
       "    u'indexTerm_title': u'Sigmoid flexure of colon'}],\n",
       "  u'is_leaf_node': False,\n",
       "  u'mms_id': u'894847549',\n",
       "  u'parents': [u'578694909'],\n",
       "  u'postcoordinationScale': [],\n",
       "  u'title': u'Sigmoid colon'},\n",
       " {u'_id': ObjectId('5d950eec0ef932347358fa72'),\n",
       "  u'childs': [],\n",
       "  u'code': u'7B02.0',\n",
       "  u'defn': u'NA',\n",
       "  u'has_children': 0,\n",
       "  u'has_parents': 1,\n",
       "  u'index_term': [{u'indexTerm_foundation_id': u'NA',\n",
       "    u'indexTerm_title': u'Hypnogogic exploding head syndrome'},\n",
       "   {u'indexTerm_foundation_id': u'NA',\n",
       "    u'indexTerm_title': u'Hypnogogic sensory disturbance'}],\n",
       "  u'is_leaf_node': False,\n",
       "  u'mms_id': u'1393951214',\n",
       "  u'parents': [u'192130563'],\n",
       "  u'postcoordinationScale': [],\n",
       "  u'title': u'Hypnogogic exploding head syndrome'}]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DATA[0:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "db = client[\"icd11\"]\n",
    "id2title_col = db[\"id2title\"]\n",
    "\n",
    "\n",
    "for x in id2title_col.find():\n",
    "    id2title = x "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "#id2title"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
